downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/BiologicalProcessAnnotationBuilder.java

```diff
@@ -1,17 +1,26 @@
 package org.reactome.release.downloadDirectory.GenerateGOAnnotationFile;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.gk.model.GKInstance;
 import org.gk.model.ReactomeJavaConstants;
 import org.gk.schema.SchemaClass;
 
 import java.util.*;
-import java.util.stream.Collectors;
 
 public class BiologicalProcessAnnotationBuilder {
 
+    private static final Logger logger = LogManager.getLogger();
+
+    // When attempting to find BiologicalProcess accessions, sometimes referral Event instances need to be checked. We cap this at 2 recursions.
     private static final int MAX_RECURSION_LEVEL = 2;
     private static final String BIOLOGICAL_PROCESS_LETTER = "P";
 
+    /**
+     * Initial Biological Function annotations method that determines how to retrieve proteins for annotation. Protein retrieval methods differ depending on the presence of a catalyst.
+     * @param reactionInst
+     * @throws Exception
```

Comment:
For the JavaDoc annotations of parameters, exceptions, and return values, could a description be added for what the parameter is, when the exception will be thrown, and what the value being returned is, respectively?

For public methods, the JavaDoc should describe everything going into and coming out of a function and under what circumstances in throws exceptions.